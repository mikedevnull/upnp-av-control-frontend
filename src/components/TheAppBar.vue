<template>
  <header class="mdc-top-app-bar">
    <div class="mdc-top-app-bar__row">
      <section class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start">
        <span class="mdc-top-app-bar__title">UPnP AV controller</span>
      </section>
    </div>
  </header>
</template>

<script>
import { MDCTopAppBar } from "@material/top-app-bar";

export default {
  components: {},
  computed: {},
  data() {
    return {
      mdcAppBar: undefined
    };
  },
  mounted() {
    this.mdcAppBar = MDCTopAppBar.attachTo(this.$el);
  },
  beforeDestroy() {
    if (typeof this.mdcAppBar !== "undefined") {
      this.mdcAppBar.destroy();
    }
  }
};
</script>
<style lang="scss">
@use '@material/top-app-bar/mdc-top-app-bar';
</style>
