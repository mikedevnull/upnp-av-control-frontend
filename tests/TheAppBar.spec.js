import { mount, RouterLinkStub } from '@vue/test-utils'
import TheAppBar from '@/components/TheAppBar.vue'


describe('TheAppBar.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(TheAppBar, { stubs: { RouterLink: RouterLinkStub } });
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  })
})
