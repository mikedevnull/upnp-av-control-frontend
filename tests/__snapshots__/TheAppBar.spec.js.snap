// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheAppBar.vue renders correctly 1`] = `
<header
  class="mdc-top-app-bar"
>
  <div
    class="mdc-top-app-bar__row"
  >
    <section
      class="mdc-top-app-bar__section mdc-top-app-bar__section--align-start"
    >
      <span
        class="mdc-top-app-bar__title"
      >
        UPnP AV controller
      </span>
    </section>
  </div>
</header>
`;
